### ISSUE TITLE
v7.0.7: MDF concatenation does not work if timestamps overlap

### ISSUE THREAD

**Issue by MatinF** [danielhrisca/asammdf#705]
Hi Daniel,

I've done some tests with MDF log files from 2 CANedge devices that record data in parallel.

If I try to concatenate raw CAN bus data from these two devices in asammdf, the expected result would be that their original timestamps would be retained. For example, in my test I log data for both devices over the period 17:26 to 17:32 and would thus expect the concatenated log file to also cover 17:26 to 17:32.

However, in practice, asammdf puts the log files in sequence so that the concatenated data covers 17:26 to 17:38 - with the log files being fully separated.

Is it possible to modify this behavior so that the timestamps are correctly injected when files are concatenated like this? I am not sure if this has drastic negative speed implications, but currently it is not e.g. possible to "merge" data from multiple devices into one.

Martin

**Comment by danielhrisca** [comment]
If they are measured in parallel then you want to use the ``stack`` method (https://asammdf.readthedocs.io/en/latest/api.html#asammdf.mdf.MDF.stack)

**Comment by MatinF** [comment]
Thanks for the tip Daniel. It does seem to make sense in terms of what it produces for the raw CAN bus data. However, if I then try to DBC decode the stacked file, I get an error as below when trying to plot any of the signals:
![image](https://user-images.githubusercontent.com/26184427/169710641-6c2163ff-9cf4-473d-9125-0df823e73bd4.png)

**Comment by danielhrisca** [comment]
The CAN bus channel number is probably the same in the two files.

**Comment by MatinF** [comment]
Yes, the data in both files are across both Bus Channel 1 and 2. But since it looks to make sense for the raw data, I'm not sure why that would necessarily cause the DBC decoding to fail?

**Comment by danielhrisca** [comment]
The DBC decode will place decodes data from both files in the same channel. This means that the decoded data from the second measurement will actually get appended to the decoded data from the first measurement , with the result that the timestamp will not be monotonously increasing

**Comment by MatinF** [comment]
Ok would it be an option in that case to pre-sort the stacked file by timestamps before performing the actual DBC decoding? Or maybe I'm missing a factor that would make this problematic

**Comment by MatinF** [comment]
Closing this as I have not heard further from the customer
